src/mumble/ServerHandler.cpp

```diff
@@ -426,8 +426,10 @@ void ServerHandler::run() {
 
 	#ifdef Q_OS_WIN
 			if (hQoS) {
-				if (! QOSRemoveSocketFromFlow(hQoS, 0, dwFlowUDP, 0))
-					qWarning("ServerHandler: Failed to remove UDP from QoS");
+				if (!QOSRemoveSocketFromFlow(hQoS, 0, dwFlowUDP, 0)) {
+					qWarning("ServerHandler: Failed to remove UDP from QoS. QOSRemoveSocketFromFlow() failed with error %u!", GetLastError());
```

Comment:
Doesn't `GetLastError` return a string? I'm that case I think `%u` might be wrong :thinking:

Reply:
It returns a `DWORD`.

https://docs.microsoft.com/en-us/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror

Reply:
And `%u` is suitable for that? At forts glance I kinda expected this to stand for unsigned numbers or something like that...

Reply:
From https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types:

> **DWORD**
A 32-bit unsigned integer. The range is 0 through 4294967295 decimal.
This type is declared in IntSafe.h as follows:
`typedef unsigned long DWORD;`

Reply:
A lol - I always thought `DWORD` was some kind of string xD